### PR TITLE
Medical Scanner Buff + Qol

### PR DESCRIPTION
**Pull request by hyper2snyper**
## About The Pull Request

Med scanner is now tgui.
Larvas are now directly detected.
Fractures too.

Scanner now auto updates for corpmen and will even update within bags.
It has a range of three currently for health tracking.

image
image
image
image
image
image
Noone is allowed to point out the 'Slaved' or the fact that the brute numbers are shifted 5 pixels to the right and arent aligned. They are all fixed see

image
image

image
Now with shrapnel tracking.

## Why It's Good For The Game
The medical scanner is hard to look at and annoying to use as a corpman since you have to keep clicking the patient. This modernizes it and clears up the information it gives.

## Changelog
:cl:
qol: Med Scanner is now TGUI. 
balance: Med Scanner now auto updates for Corpmen and has a remote distance of 3
/:cl:

### CODE REVIEW

**Comment by Andrewhedge** [comment]
seal man good

**Comment by Losenis** [comment]
Does the autoupdate need the scanner held in hand or can it be put back in storage/thrown?

**Comment by hyper2snyper** [comment]
> Does the autoupdate need the scanner held in hand or can it be put back in storage/thrown?

Can be in storage. 
If it's dropped the window will stay open but it will stop refreshing.
If you move out of range of the patient then it'll stop refreshing as well

**Comment by Lumipharon** [comment]
To clarify, you can always see exactly where fracs are with this change?

**Comment by Arctrooper8** [comment]
And what about shrapnel too? You said in discord you'd just show where it was. Is that still true?

**Comment by hyper2snyper** [comment]
> And what about shrapnel too? You said in discord you'd just show where it was. Is that still true?

oh right i forgot about that. well currently it shows that there is shrapnel and tells how much, but it doesnt show exactly where

**Comment by hyper2snyper** [comment]
> To clarify, you can always see exactly where fracs are with this change?

yes

File: code/game/objects/items/scanners.dm [hunk 1]
```diff
@@ -67,404 +72,181 @@ REAGENT SCANNER
...
+/obj/item/healthanalyzer/attack(mob/living/carbon/M, mob/living/user)
+	. = ..()
 	if(user.skills.getRating("medical") < skill_threshold)
 		to_chat(user, span_warning("You start fumbling around with [src]..."))
-		var/fduration = max(SKILL_TASK_AVERAGE - (1 SECONDS * user.skills.getRating("medical")), 0)
-		if(!do_mob(user, M, fduration, BUSY_ICON_UNSKILLED))
+		if(!do_mob(user, M, max(SKILL_TASK_AVERAGE - (1 SECONDS * user.skills.getRating("medical")), 0), BUSY_ICON_UNSKILLED))
 			return
+	playsound(src.loc, 'sound/items/healthanalyzer.ogg', 50)
+	if(!iscarbon(M) || issynth(M))
```
[the remark this answers is not in the document]
Comment (Arctrooper8) [reply code/game/objects/items/scanners.dm:93]: We should just remove the no scan trait for Synth, you can scan combat robots and they're basically just synth that can use guns.

**Comment by Losenis** [comment]
How does this show IB damage?

**Comment by bowlofcereal** [comment]
Does this specify that a certain limb has an implant in it and NOT shrapnel, so the medic doesnt yank it out and do significant amounts of damage and ruin the implant?

**Comment by Pariah919** [comment]
QOL is good, a +1 from me.

**Comment by bluezorua** [comment]
does this also apply/work for armour attachments that add internal analysers like Valk/Vali?

**Comment by Losenis** [comment]
Doesn't show IB nor Necrotizing, needs it asap

**Comment by hyper2snyper** [comment]
> Doesn't show IB nor Necrotizing, needs it asap

It does show both

**Comment by ilikesaying** [comment]
image

**Comment by Losenis** [comment]
It doesn't show IB in that it only shows brute damage that isn't healing, I haven't seen anything that'd point to it being actual IB other than the player already knowing that brute damage that refuses to heal is IB, unlike old scanner that says there's IB somewhere.

**Review by BraveMole — CHANGES_REQUESTED** [review]
You can use scanner on you, throw it away, and you get infini health information. Please fix. Not having in your hand should stop the info feed

**Comment by FosterGrime-Heymynameslouie** [comment]
I know when you bleed

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**